src/SourceGenerators/Uno.UI.SourceGenerators.Tests/Given_PathMarkupParser.cs

```diff
@@ -227,6 +227,8 @@ public void CloseFigure_Should_Move_CurrentPoint_To_CreateFigurePoint()
 }, global::Windows.UI.Xaml.Media.FillRule.EvenOdd)", generatedCode);
 		}
 
+#if !NET6_0_OR_GREATER
```

Comment:
Curious why this is needed?

Reply:
This is the only test that failed in NET6. I didn't look into the reasons.

Reply:
I think something related to the NET461 constants in the SourceGenerators project

Reply:
What was the actual result in .NET 6?

Reply:
```
 Boolean_Should_Be_Read_From_The_Next_Non_Whitespace_Character_Only
   Source: Given_PathMarkupParser.cs line 231
   Duration: 46 ms

  Message: 
Assert.AreEqual failed. Expected:<global::Uno.Media.GeometryHelper.Build(c =>
{
c.BeginFigure(new global::Windows.Foundation.Point(17.432, 11.619), true);
c.BezierTo(new global::Windows.Foundation.Point(17.456, 11.701), new global::Windows.Foundation.Point(17.472, 11.784), new global::Windows.Foundation.Point(17.472, 11.866), true, false);
c.LineTo(new global::Windows.Foundation.Point(17.472, 26.54), true, false);
c.BezierTo(new global::Windows.Foundation.Point(17.472, 26.91), new global::Windows.Foundation.Point(17.201, 27.092), new global::Windows.Foundation.Point(16.867, 26.949), true, false);
c.LineTo(new global::Windows.Foundation.Point(11.528, 24.667), true, false);
c.BezierTo(new global::Windows.Foundation.Point(11.192, 24.523), new global::Windows.Foundation.Point(10.924, 24.109), new global::Windows.Foundation.Point(10.924, 23.741), true, false);
c.LineTo(new global::Windows.Foundation.Point(10.924, 9.066), true, false);
c.BezierTo(new global::Windows.Foundation.Point(10.924, 8.698), new global::Windows.Foundation.Point(11.194, 8.515), new global::Windows.Foundation.Point(11.528, 8.657), true, false);
c.LineTo(new global::Windows.Foundation.Point(16.867, 10.94), true, false);
c.ArcTo(new global::Windows.Foundation.Point(17.137, 11.128), new global::Windows.Foundation.Size(0.898, 0.898), 0d, false, global::Windows.UI.Xaml.Media.SweepDirection.Clockwise, true, false);
c.BezierTo(new global::Windows.Foundation.Point(17.227, 11.297), new global::Windows.Foundation.Point(17.326, 11.461), new global::Windows.Foundation.Point(17.432, 11.618), true, false);
c.SetClosedState(false);
c.BeginFigure(new global::Windows.Foundation.Point(9.615, 9.07), true);
c.LineTo(new global::Windows.Foundation.Point(9.615, 23.745), true, false);
c.BezierTo(new global::Windows.Foundation.Point(9.615, 24.113), new global::Windows.Foundation.Point(9.345, 24.527), new global::Windows.Foundation.Point(9.01, 24.67), true, false);
c.LineTo(new global::Windows.Foundation.Point(3.671, 26.952), true, false);
c.BezierTo(new global::Windows.Foundation.Point(3.337, 27.095), new global::Windows.Foundation.Point(3.067, 26.912), new global::Windows.Foundation.Point(3.067, 26.544), true, false);
c.LineTo(new global::Windows.Foundation.Point(3.067, 11.868), true, false);
c.BezierTo(new global::Windows.Foundation.Point(3.067, 11.5), new global::Windows.Foundation.Point(3.336, 11.086), new global::Windows.Foundation.Point(3.671, 10.942), true, false);
c.LineTo(new global::Windows.Foundation.Point(9.011, 8.66), true, false);
c.BezierTo(new global::Windows.Foundation.Point(9.344, 8.517), new global::Windows.Foundation.Point(9.615, 8.7), new global::Windows.Foundation.Point(9.615, 9.07), true, false);
c.SetClosedState(false);
c.BeginFigure(new global::Windows.Foundation.Point(25.328, 13.243), true);
c.LineTo(new global::Windows.Foundation.Point(25.328, 23.74), true, false);
c.BezierTo(new global::Windows.Foundation.Point(25.328, 24.108), new global::Windows.Foundation.Point(25.058, 24.522), new global::Windows.Foundation.Point(24.723, 24.666), true, false);
c.LineTo(new global::Windows.Foundation.Point(19.385, 26.948), true, false);
c.BezierTo(new global::Windows.Foundation.Point(19.051, 27.091), new global::Windows.Foundation.Point(18.781, 26.908), new global::Windows.Foundation.Point(18.781, 26.538), true, false);
c.LineTo(new global::Windows.Foundation.Point(18.781, 13.216), true, false);
c.BezierTo(new global::Windows.Foundation.Point(19.796, 14.447), new global::Windows.Foundation.Point(21.483, 16.831), new global::Windows.Foundation.Point(21.917, 19.516), true, false);
c.LineTo(new global::Windows.Foundation.Point(22.229, 19.516), true, false);
c.BezierTo(new global::Windows.Foundation.Point(22.659, 16.851), new global::Windows.Foundation.Point(24.316, 14.483), new global::Windows.Foundation.Point(25.328, 13.244), true, false);
c.SetClosedState(false);
c.BeginFigure(new global::Windows.Foundation.Point(22.111, 10.854), true);
c.BezierTo(new global::Windows.Foundation.Point(20.046, 10.854), new global::Windows.Foundation.Point(18.373, 9.149), new global::Windows.Foundation.Point(18.373, 7.046), true, false);
c.BezierTo(new global::Windows.Foundation.Point(18.373, 4.944), new global::Windows.Foundation.Point(20.046, 3.239), new global::Windows.Foundation.Point(22.111, 3.239), true, false);
c.BezierTo(new global::Windows.Foundation.Point(24.175, 3.239), new global::Windows.Foundation.Point(25.849, 4.944), new global::Windows.Foundation.Point(25.849, 7.046), true, false);
c.BezierTo(new global::Windows.Foundation.Point(25.849, 9.149), new global::Windows.Foundation.Point(24.175, 10.854), new global::Windows.Foundation.Point(22.111, 10.854), true, false);
c.SetClosedState(false);
c.BeginFigure(new global::Windows.Foundation.Point(22.054, 2), true);
c.BezierTo(new global::Windows.Foundation.Point(19.286, 2), new global::Windows.Foundation.Point(17.042, 4.286), new global::Windows.Foundation.Point(17.042, 7.105), true, false);
c.BezierTo(new global::Windows.Foundation.Point(17.042, 8.483), new global::Windows.Foundation.Point(17.573, 9.798), new global::Windows.Foundation.Point(18.443, 10.716), true, false);
c.BezierTo(new global::Windows.Foundation.Point(18.443, 10.716), new global::Windows.Foundation.Point(21.371, 13.628), new global::Windows.Foundation.Point(21.931, 17.105), true, false);
c.LineTo(new global::Windows.Foundation.Point(22.21, 17.105), true, false);
c.BezierTo(new global::Windows.Foundation.Point(22.77, 13.628), new global::Windows.Foundation.Point(25.681, 10.716), new global::Windows.Foundation.Point(25.681, 10.716), true, false);
c.BezierTo(new global::Windows.Foundation.Point(26.554, 9.798), new global::Windows.Foundation.Point(27.067, 8.484), new global::Windows.Foundation.Point(27.067, 7.106), true, false);
c.BezierTo(new global::Windows.Foundation.Point(27.067, 4.286), new global::Windows.Foundation.Point(24.822, 2), new global::Windows.Foundation.Point(22.054, 2), true, false);
c.SetClosedState(false);
}, global::Windows.UI.Xaml.Media.FillRule.EvenOdd)>. Actual:<global::Uno.Media.GeometryHelper.Build(c =>
{
c.BeginFigure(new global::Windows.Foundation.Point(17.432, 11.619), true);
c.BezierTo(new global::Windows.Foundation.Point(17.456, 11.701), new global::Windows.Foundation.Point(17.471999999999998, 11.783999999999999), new global::Windows.Foundation.Point(17.471999999999998, 11.866), true, false);
c.LineTo(new global::Windows.Foundation.Point(17.471999999999998, 26.54), true, false);
c.BezierTo(new global::Windows.Foundation.Point(17.471999999999998, 26.91), new global::Windows.Foundation.Point(17.200999999999997, 27.092), new global::Windows.Foundation.Point(16.866999999999997, 26.948999999999998), true, false);
c.LineTo(new global::Windows.Foundation.Point(11.527999999999997, 24.666999999999998), true, false);
c.BezierTo(new global::Windows.Foundation.Point(11.191999999999997, 24.523), new global::Windows.Foundation.Point(10.923999999999998, 24.108999999999998), new global::Windows.Foundation.Point(10.923999999999998, 23.741), true, false);
c.LineTo(new global::Windows.Foundation.Point(10.923999999999998, 9.066), true, false);
c.BezierTo(new global::Windows.Foundation.Point(10.923999999999998, 8.698), new global::Windows.Foundation.Point(11.193999999999997, 8.515), new global::Windows.Foundation.Point(11.527999999999997, 8.657), true, false);
c.LineTo(new global::Windows.Foundation.Point(16.866999999999997, 10.94), true, false);
c.ArcTo(new global::Windows.Foundation.Point(17.136999999999997, 11.128), new global::Windows.Foundation.Size(0.898, 0.898), 0d, false, global::Windows.UI.Xaml.Media.SweepDirection.Clockwise, true, false);
c.BezierTo(new global::Windows.Foundation.Point(17.226999999999997, 11.297), new global::Windows.Foundation.Point(17.325999999999997, 11.461), new global::Windows.Foundation.Point(17.432, 11.618), true, false);
c.SetClosedState(false);
c.BeginFigure(new global::Windows.Foundation.Point(9.615, 9.07), true);
c.LineTo(new global::Windows.Foundation.Point(9.615, 23.745), true, false);
c.BezierTo(new global::Windows.Foundation.Point(9.615, 24.113), new global::Windows.Foundation.Point(9.345, 24.527), new global::Windows.Foundation.Point(9.01, 24.67), true, false);
c.LineTo(new global::Windows.Foundation.Point(3.6709999999999994, 26.952), true, false);
c.BezierTo(new global::Windows.Foundation.Point(3.3369999999999993, 27.095000000000002), new global::Windows.Foundation.Point(3.0669999999999993, 26.912000000000003), new global::Windows.Foundation.Point(3.0669999999999993, 26.544), true, false);
c.LineTo(new global::Windows.Foundation.Point(3.0669999999999993, 11.868), true, false);
c.BezierTo(new global::Windows.Foundation.Point(3.0669999999999993, 11.5), new global::Windows.Foundation.Point(3.3359999999999994, 11.086), new global::Windows.Foundation.Point(3.6709999999999994, 10.942), true, false);
c.LineTo(new global::Windows.Foundation.Point(9.011, 8.66), true, false);
c.BezierTo(new global::Windows.Foundation.Point(9.344, 8.517), new global::Windows.Foundation.Point(9.614999999999998, 8.7), new global::Windows.Foundation.Point(9.614999999999998, 9.07), true, false);
c.SetClosedState(false);
c.BeginFigure(new global::Windows.Foundation.Point(25.327999999999996, 13.243), true);
c.LineTo(new global::Windows.Foundation.Point(25.327999999999996, 23.74), true, false);
c.BezierTo(new global::Windows.Foundation.Point(25.327999999999996, 24.107999999999997), new global::Windows.Foundation.Point(25.057999999999996, 24.522), new global::Windows.Foundation.Point(24.722999999999995, 24.665999999999997), true, false);
c.LineTo(new global::Windows.Foundation.Point(19.384999999999994, 26.947999999999997), true, false);
c.BezierTo(new global::Windows.Foundation.Point(19.050999999999995, 27.090999999999998), new global::Windows.Foundation.Point(18.780999999999995, 26.907999999999998), new global::Windows.Foundation.Point(18.780999999999995, 26.537999999999997), true, false);
c.LineTo(new global::Windows.Foundation.Point(18.780999999999995, 13.216), true, false);
c.BezierTo(new global::Windows.Foundation.Point(19.795999999999996, 14.447), new global::Windows.Foundation.Point(21.482999999999997, 16.831), new global::Windows.Foundation.Point(21.916999999999994, 19.516), true, false);
c.LineTo(new global::Windows.Foundation.Point(22.228999999999996, 19.516), true, false);
c.BezierTo(new global::Windows.Foundation.Point(22.658999999999995, 16.851), new global::Windows.Foundation.Point(24.315999999999995, 14.482999999999997), new global::Windows.Foundation.Point(25.327999999999996, 13.243999999999998), true, false);
c.SetClosedState(false);
c.BeginFigure(new global::Windows.Foundation.Point(22.110999999999997, 10.853999999999997), true);
c.BezierTo(new global::Windows.Foundation.Point(20.045999999999996, 10.853999999999997), new global::Windows.Foundation.Point(18.372999999999998, 9.148999999999997), new global::Windows.Foundation.Point(18.372999999999998, 7.045999999999998), true, false);
c.BezierTo(new global::Windows.Foundation.Point(18.372999999999998, 4.943999999999997), new global::Windows.Foundation.Point(20.046, 3.2389999999999977), new global::Windows.Foundation.Point(22.110999999999997, 3.2389999999999977), true, false);
c.BezierTo(new global::Windows.Foundation.Point(24.174999999999997, 3.2389999999999977), new global::Windows.Foundation.Point(25.848999999999997, 4.943999999999997), new global::Windows.Foundation.Point(25.848999999999997, 7.045999999999998), true, false);
c.BezierTo(new global::Windows.Foundation.Point(25.848999999999997, 9.148999999999997), new global::Windows.Foundation.Point(24.174999999999997, 10.853999999999997), new global::Windows.Foundation.Point(22.110999999999997, 10.853999999999997), true, false);
c.SetClosedState(false);
c.BeginFigure(new global::Windows.Foundation.Point(22.054, 2), true);
c.BezierTo(new global::Windows.Foundation.Point(19.285999999999998, 2), new global::Windows.Foundation.Point(17.041999999999998, 4.286), new global::Windows.Foundation.Point(17.041999999999998, 7.105), true, false);
c.BezierTo(new global::Windows.Foundation.Point(17.041999999999998, 8.483), new global::Windows.Foundation.Point(17.572999999999997, 9.798), new global::Windows.Foundation.Point(18.442999999999998, 10.716000000000001), true, false);
c.BezierTo(new global::Windows.Foundation.Point(18.442999999999998, 10.716000000000001), new global::Windows.Foundation.Point(21.371, 13.628), new global::Windows.Foundation.Point(21.930999999999997, 17.105), true, false);
c.LineTo(new global::Windows.Foundation.Point(22.209999999999997, 17.105), true, false);
c.BezierTo(new global::Windows.Foundation.Point(22.769999999999996, 13.628), new global::Windows.Foundation.Point(25.680999999999997, 10.716000000000001), new global::Windows.Foundation.Point(25.680999999999997, 10.716000000000001), true, false);
c.BezierTo(new global::Windows.Foundation.Point(26.554, 9.798000000000002), new global::Windows.Foundation.Point(27.066999999999997, 8.484000000000002), new global::Windows.Foundation.Point(27.066999999999997, 7.106000000000002), true, false);
c.BezierTo(new global::Windows.Foundation.Point(27.066999999999997, 4.286000000000001), new global::Windows.Foundation.Point(24.821999999999996, 2.0000000000000018), new global::Windows.Foundation.Point(22.053999999999995, 2.0000000000000018), true, false);
c.SetClosedState(false);
}, global::Windows.UI.Xaml.Media.FillRule.EvenOdd)>. 
```

Reply:
This is where Snapshot testing would look good (via https://github.com/VerifyTests/Verify.SourceGenerators) because it outputs a comparison like in git

Reply:
<img width="1596" alt="image" src="https://user-images.githubusercontent.com/3002068/163592115-a22900c1-3b95-4549-9828-55b7314b3b74.png">
We need to add "{:F3}" or something like this to DrawingExtensions.ToCode

Reply:
The difference is in floating point numbers. For example, `17.471999999999998` vs `17.472`.

I think something has changed in the .NET runtime causing this small difference.

Reply:
I don't think it's worth doing changes in product code. Either keep the test ignored in .NET 6, or assert difference results per the runtime, or update the numbers used as test input to something that doesn't cause those differences.

Reply:
Let's do two assertions, one for each target, just as a matter of detecting regression (assuming that the behavior is the right one for now).